src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/JpaActionPlanPersistenceAdapterTest.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.kotlin.any
+import org.mockito.kotlin.given
+import org.mockito.kotlin.verify
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.aValidPrisonNumber
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.entity.aValidActionPlanEntity
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.mapper.ActionPlanEntityMapper
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa.repository.ActionPlanRepository
+import uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal.aValidActionPlan
+
+@ExtendWith(MockitoExtension::class)
+class JpaActionPlanPersistenceAdapterTest {
+
+  @InjectMocks
+  private lateinit var persistenceAdapter: JpaActionPlanPersistenceAdapter
+
+  @Mock
+  private lateinit var actionPlanRepository: ActionPlanRepository
+
+  @Mock
+  private lateinit var actionPlanMapper: ActionPlanEntityMapper
+
+  @Test
+  fun `should retrieve Action Plan`() {
+    // Given
+    val prisonNumber = aValidPrisonNumber()
+    val actionPlanEntity = aValidActionPlanEntity(prisonNumber = prisonNumber)
+    val expectedActionPlanDomain = aValidActionPlan(prisonNumber = prisonNumber)
+    given(actionPlanRepository.findByPrisonNumber(any())).willReturn(actionPlanEntity)
+    given(actionPlanMapper.fromEntityToDomain(any())).willReturn(expectedActionPlanDomain)
+
+    // When
+    val actual = persistenceAdapter.getActionPlan(prisonNumber)
+
+    // Then
+    assertThat(actual).isEqualTo(expectedActionPlanDomain)
+    verify(actionPlanRepository).findByPrisonNumber(prisonNumber)
+    verify(actionPlanMapper).fromEntityToDomain(actionPlanEntity)
+  }
```

Comment:
Do we need a test for when the action plan cannot be found, and the adapter throws the exception?

(Though I'm not convinced about the persistence adapter throwing exceptions .... we might need to think that one through)

Reply:
Good point. I was thinking we want an exception to be thrown as early as possible, but I guess there could be a use case where `null` is acceptable (for example when checking if the entity exists). I'll move it to the domain service and throw the exception there instead.

Reply:
Moved to `ActionPlanService` (and corresponding test added).